run.cc

```diff
@@ -68,12 +73,60 @@ int main() {
   cout << "******************************************************" << endl;
   // create an object which holds data of experiment A
   Data datA("exp_A");
+  Data datB("exp_B");
+  Data datC("exp_C");
+  Data datD("exp_D");
+  std::vector<Data> data;
+  data.push_back(datA);
+  data.push_back(datB);
+  data.push_back(datC);
+  data.push_back(datD);
+  
+ 
+
 
   // here is the data from experiment A
   cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
        << endl;
   cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
        << endl;
+  cout << "measurement of experiment B in bin 27: " << datB.measurement(27)
+       << endl;
+  cout << "measurement of experiment C in bin 27: " << datC.measurement(27)
+       << endl;
+  cout << "measurement of experiment D in bin 27: " << datD.measurement(27)
+       << endl;
+
+  double uncert_A = datA.error(27);
+  double uncert_B = datB.error(27);
+
+  cout << "n*uncertainty " << 2*sqrt(pow(uncert_A,2)+pow(uncert_B,2)) << endl;
+  cout << "delta Y " << abs(datA.measurement(27)-datB.measurement(27)) << endl;
+
+  int count = datA.checkCompatibility(datD, 1);
+  cout << "count: " << count << endl;
+  Data datAB = datA.averageDatasets(datB);
+  Data datCD = datC.averageDatasets(datD);
+  Data datABCD = datAB.averageDatasets(datCD);
+  
+  cout << "for datA, " << datA.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datB, " << datB.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datC, " << datC.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datD, " << datD.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+
+  cout << "for datA, " << datA.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datB, " << datB.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datC, " << datC.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datD, " << datD.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+
```

Comment:
In 2a) you didn't answer how many deviations you would typically expect even if the data were compatible. If you draw two samples from the same distribution, you would still expect some variation and even the probability to see a datapoint that has a deviation of 3 sigma is non-zero. So what you should have done here is just plug in the gaussian probability outside of 3 sigma (and 2 sigma) and multiply this with the number of datapoints, then see if the number of actual deviations agrees with what you computed.  Since your print out is correct, you'll still get 0.5 points for 2a)